src/aead.rs

```diff
@@ -664,6 +664,76 @@ enum Direction {
     Sealing,
 }
 
+trait BitXor {
+    fn bitxor_assign(&mut self, input: [u8; BLOCK_LEN]);
+}
+
+impl BitXor for [u8; BLOCK_LEN] {
+    fn bitxor_assign(&mut self, input: [u8; BLOCK_LEN]) {
+        for (r, a) in self.iter_mut().zip(input.iter()) {
+            *r ^= *a;
+        }
+    }
+}
+
+trait ConvertEndian {
+    fn u64s_be_to_native(&self) -> [u64; 2];
+    fn from_be_u64s(u64s: [u64; 2]) -> Self;
```

Comment:
I think the right thing to do is to move all the wierd endianness logic for GCM into the GCM submodule, and all the wierd endianness logic for AES into the AES submodule, with the hope that we can then simplify and remove a lot of the wierdness. In particular, I don't think this higher-level `aead` module should be exposing this wierdness to all its submodules, because our goal should be to eliminate the weirdness.

Reply:
> I think the right thing to do is to move all the wierd endianness logic for GCM into the GCM submodule...

Yeah, wasn't sure where to put the endianness stuff, but that definitely makes sense.

> our goal should be to eliminate the weirdness

Unweird all the things! :)